CatalogReaderExample/NuGet.Protocol.Catalog/Models/ModelExtensions.cs

```diff
@@ -0,0 +1,168 @@
+            }
+        }
+
+        /// <summary>
+        /// Parse the package version as a <see cref="NuGetVersion" />.
+        /// </summary>
+        /// <param name="leaf">The catalog leaf.</param>
+        /// <returns>The package version.</returns>
+        public static NuGetVersion ParsePackageVersion(this ICatalogLeafItem leaf)
+        {
+            return NuGetVersion.Parse(leaf.PackageVersion);
+        }
+
+        /// <summary>
+        /// Parse the target framework as a <see cref="NuGetFramework" />.
+        /// </summary>
+        /// <param name="packageDependencyGroup">The package dependency group.</param>
+        /// <returns>The framework.</returns>
+        public static NuGetFramework ParseTargetFramework(this PackageDependencyGroup packageDependencyGroup)
+        {
+            if (string.IsNullOrEmpty(packageDependencyGroup.TargetFramework))
+            {
+                return NuGetFramework.AnyFramework;
+            }
+
+            return NuGetFramework.Parse(packageDependencyGroup.TargetFramework);
+        }
+
+        /// <summary>
+        /// Parse the version range as a <see cref="VersionRange"/>.
+        /// </summary>
+        /// <param name="packageDependency">The package dependency.</param>
+        /// <returns>The version range.</returns>
+        public static VersionRange ParseRange(this PackageDependency packageDependency)
+        {
+            if (string.IsNullOrEmpty(packageDependency.Range))
+            {
+                return VersionRange.All;
+            }
+
+            return VersionRange.Parse(packageDependency.Range);
+        }
+
+        /// <summary>
+        /// Determines if the provided catalog leaf is a package delete.
+        /// </summary>
+        /// <param name="leaf">The catalog leaf.</param>
+        /// <returns>True if the catalog leaf represents a package delete.</returns>
+        public static bool IsPackageDelete(this ICatalogLeafItem leaf)
+        {
+            return leaf.Type == CatalogLeafType.PackageDelete;
+        }
+
+        /// <summary>
+        /// Determines if the provided catalog leaf is contains package details.
+        /// </summary>
+        /// <param name="leaf">The catalog leaf.</param>
+        /// <returns>True if the catalog leaf contains package details.</returns>
+        public static bool IsPackageDetails(this ICatalogLeafItem leaf)
+        {
+            return leaf.Type == CatalogLeafType.PackageDetails;
+        }
+
+        /// <summary>
+        /// Determines if the provided package details list represents a listed package.
+        /// </summary>
+        /// <param name="leaf">The catalog leaf.</param>
+        /// <returns>True if the package is listed.</returns>
+        public static bool IsListed(this PackageDetailsCatalogLeaf leaf)
+        {
+            if (leaf.Listed.HasValue)
+            {
+                return leaf.Listed.Value;
+            }
+
+            return leaf.Published.Year == 1900;
```

Comment:
Oh, I didn't know this was a thing. A comment may be good here! :)

Reply:
This is documented in the API docs, but I will include it here for people building/contributing to this SDK.

Reply:
Fixed.